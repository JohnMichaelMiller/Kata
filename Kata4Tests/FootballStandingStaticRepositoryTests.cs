using Microsoft.VisualStudio.TestTools.UnitTesting;
using Kata4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace Kata4.Tests
{
    [TestClass()]
    [UseReporter(typeof(DiffReporter))]
    public class FootballStandingStaticRepositoryTests
    {
        [TestMethod()]
        public void FootballStandingStaticRepositoryInstantiates()
        {
            var repo = new FootballStandingStaticRepository();
            Approvals.VerifyAll(repo, "Football Standing");
        }

        [TestMethod()]
        public void FootballStandingStaticRepositoryHasCorrectMinimumGoalDifferential()
        {
            var repo = new FootballStandingStaticRepository();
            var result =
                from w in repo
                where -w.GoalDifferential ==
                    repo.Min(d => -d.GoalDifferential)
                select w;
                         
            Approvals.VerifyAll(result, "Football Standing");
        }
    }
}