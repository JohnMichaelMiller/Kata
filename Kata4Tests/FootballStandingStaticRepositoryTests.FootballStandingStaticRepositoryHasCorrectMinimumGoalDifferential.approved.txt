Football Standing[0] = Rank:5 Team: Leeds, Goals For 53, Goals Against: 37, Goal Diff: 16
